Automattic/sensei
# Add usage tracking

Implements https://github.com/Automattic/sensei/issues/1958.

--- Additions by @alexsanford ---

## Comments

- ~The opt-in dialog is simple and functional, but the UX isn't great. Currently, it redirects the user to the Settings page and displays a "Success" notice. Probably it should be done over AJAX instead.~

## Testing

- Upon activating the plugin, the biweekly job for sending tracking data should have been scheduled.
  - Check to ensure the job is scheduled by inspecting the Cron schedule ([this plugin](https://wordpress.org/plugins/wp-crontrol/) is handy).
  - You can also use that plugin to reschedule the job to run sooner, which is useful for ensuring that the actual data gets sent to Tracks. However, the data _will not_ be sent until you have opted in (see next point).


- Ensure that the opt-in dialog and setting is working.
  - When either button is clicked, the dialog should no longer show up (to force it to show up again, run `delete_option( 'sensei_usage_tracking_opt_in_hide' );`, but note that it will only show up when the tracking option is disabled).
  - Ensure that the setting on the Sensei Settings page reflects your choice.
  - After opting in, you can now trigger the sending of data (using the [WP Crontrol](https://wordpress.org/plugins/wp-crontrol/) plugin mentioned above).


- Check [Tracks](https://mc.a8c.com/tracks/live/?eventname=sensei_stats_log) to ensure that the data was pushed. This may take a few minutes.
  - Ensure that tracking data is only sent if you have opted in to the tracking.

## TODO

- ~Need to remove the debugging page (which isn't working anyway).~

- ~Need to set a URL for the tracking policy for users to read.~
